File: api/idp.go
package api

// IdentityProviderType is the type of an identity provider.
type IdentityProviderType string

const (
	// OAuth2IdentityProvider is the identity provider type for OAuth2.
	OAuth2IdentityProvider IdentityProviderType = "OAUTH2"
	// OIDCIdentityProvider is the identity provider type for OIDC.
	OIDCIdentityProvider IdentityProviderType = "OIDC"
)

File: frontend/src/utils/idp.ts
import type { OAuth2IdentityProviderConfig } from "@/types/proto-es/v1/idp_service_pb";
import { IdentityProviderType } from "@/types/proto-es/v1/idp_service_pb";

export const identityProviderTypeToString = (
  type: IdentityProviderType
): string => {
  if (type === IdentityProviderType.OAUTH2) {
    return "OAuth 2.0";
  } else if (type === IdentityProviderType.OIDC) {
    return "OIDC";
  } else if (type == IdentityProviderType.LDAP) {
    return "LDAP";
  } else {
    throw new Error(`identity provider type ${type} not found`);
  }
};

export interface OAuth2IdentityProviderTemplate {
  title: string;
  name: string;
  domain: string;
  type: IdentityProviderType.OAUTH2;
  config: OAuth2IdentityProviderConfig;
}
